### PR TITLE
First stable version

### PR DESCRIPTION
**Pull request by g-pyxl**
Initial commit for review/testing.

- - -
This change is [Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)

### CODE REVIEW

**Review by rebeccahaines1 — CHANGES_REQUESTED** [review]
> Reviewed 22 of 55 files at r1.
> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)* status: 22 of 55 files reviewed, 5 unresolved discussions (waiting on @g-pyxl)

___
*[`README.md` line 28 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-6aBCC9VL3bg8f-ME:-OA-6aBCC9VL3bg8f-MF:b-neocex) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L28)):*
> ```Markdown
> - Integration with PanelApp for importing gene lists from predefined panels
> - Custom padding to gene regions
> - Live edit regions of interest in results viewer
> ```

I think you said you'd removed this option
___
*[`README.md` line 35 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-8I1j87GQyJBe2om1:-OA-8I1j87GQyJBe2om2:bg2eula) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L35)):*
> ```Markdown
> - BED file management system with draft and published states
> 
> ## Functionality schema
> ```

in this image there's a user action "modify" but I thought this had been removed?
___
*[`README.md` line 108 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-6wZc2qVWImVELkDB:-OA-6wZc2qVWImVELkDC:b-8gqg7r) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L108)):*
> ```Markdown
> 1. Build the Docker image:
>    ```
>    docker build -t bed-file-generator .
> ```

specify tag (for Readme could just be generic) ??
___
*[`README.md` line 113 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-74C85CwYolv1Eq6q:-OA-74C85CwYolv1Eq6r:b-4sbsqf) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L113)):*
> ```Markdown
> 2. Run the container:
>    ```
>    docker run -p 5000:5000 -e BED_GENERATOR_FLASK_KEY=<your_secret_key> bed-file-generator
> ```

again specify tag?
___
*[`README.md` line 152 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-7QHe-quF09FgOPNV:-OA-7QHe-quF09FgOPNW:bvmyvd) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L152)):*
> ```Markdown
> - Uses the IGV.js library for genomic visualization
> - Integrates with PanelApp for gene panel data
> - Utilises Ensembl VEP and TARK RESTful APIs for metadata
> ```

is there any version information for these that should be noted here?

**Review by rebeccahaines1 — CHANGES_REQUESTED** [review]
> Reviewed 25 of 55 files at r1.
> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)* status: 46 of 55 files reviewed, 13 unresolved discussions (waiting on @g-pyxl)

___
*[`README.md` line 151 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8Ab9c8r-jJpC50tmY:-OA8Ab9c8r-jJpC50tmZ:b1nilsc) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L151)):*
> ```Markdown
> - Developed by the Synnovis Bioinformatics Team @ Guy's and St Thomas' NHS Foundation Trust (2024)
> - Uses the IGV.js library for genomic visualization
> - Integrates with PanelApp for gene panel data
> ```

also version info for the panelapp API?
___
*[`app/bed_generator/api.py` line 207 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8E2B20cSKO7Y5LltS:-OA8E2B20cSKO7Y5LltT:b-6uj4jh) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/api.py#L207)):*
> ```Python
>                 return [max(matching_grch37, key=lambda x: int(x['stable_id_version']))]
> 
>     # If no MANE transcripts or matching GRCh37 transcript, selects the transcript with the highest version number.
> ```

not sure about this logic (not that the code is wrong, but that this isn't necessarily the most appropriate choice). I think if the transcript hasn't been provided and there's no MANE this needs to be a decision made by a clinical scientist. As a minimum I think any transcripts selected by this logic should be flagged?
___
*[`app/bed_generator/api.py` line 370 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8IcZqCke_D3EmZ8jw:-OA8IcZqCke_D3EmZ8jx:b7pjvpm) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/api.py#L370)):*
> ```Python
>         if len(unknown_features) > 1:
>             for feature in unknown_features:
>                 feature['alert'] = f"Coordinate {coord} overlaps multiple unknown genes."
> ```

what would this actually mean? how can a feature overlap "multiple unknown genes"?
___
*[`app/bed_generator/api.py` line 413 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8JJY_-bMpIUa3Vp3k:-OA8JJY_-bMpIUa3Vp3l:b8lowc9) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/api.py#L413)):*
> ```Python
>                 'version': panel['version'],
>                 'version_created': panel['version_created'],
>                 'genes': fetch_genes_for_panel(panel['id'], include_amber=True, include_red=True)
> ```

does the user have the option to exclude amber and red genes?
___
*[`app/bed_generator/bed_generator.py` line 32 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8KOdy-klIDG1ietdc:-OA8KOdy-klIDG1ietdd:b-q7rlzu) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/bed_generator.py#L32)):*
> ```Python
>         """
>         loc_region = f"chr{r['loc_region']}" if add_chr_prefix else r['loc_region']
>         loc_start = r['loc_start'] - padding
> ```

Just want to check that this is fine for reverse strand genes, i.e. that elsewhere the code is handling that start and end in that context might be flipped so needs to be handled first?
___
*[`app/bed_generator/routes.py` line 95 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA9U-d052pX3eVCX7SB:-OA9U-d052pX3eVCX7SC:b-9q370d) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/routes.py#L95)):*
> ```Python
>     print("Results:", results)
>     print("assembly:", assembly)
>     print("initial_query:", initial_query)  # Debug print
> ```

is this debug statement still needed?
___
*[`app/bed_generator/routes.py` line 120 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA9UGxn6Dk4INhMv4TY:-OA9UGxn6Dk4INhMv4TZ:b-khi0x9) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/routes.py#L120)):*
> ```Python
>     # Recalculate loc_start and loc_end based on new padding
>     for result in results:
>         result['loc_start'] = int(result['original_loc_start']) - padding_5
> ```

same comment as before re padding- assume reverse strand gene start and stop coords are handled so Start isn't > than stop?
___
*[`app/bed_generator/routes.py` line 197 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA9X2Rt04BnT890Ssec:-OA9X2Rt04BnT890Ssed:bxhyz0s) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/routes.py#L197)):*
> ```Python
> @bed_generator_bp.route('/submit_for_review', methods=['POST'])
> def submit_for_review():
>     try:
> ```

This is a very long try/except statement which is a bit hard to follow, especially with all the next ifs/fors. If it's working I'm happy for it to be left for now, but could an issue be added to the repo to consider reviewing/refining this in a future update?

**Review by g-pyxl — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)* status: 37 of 56 files reviewed, 13 unresolved discussions (waiting on @rebeccahaines1)

___
*[`README.md` line 28 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-6aBCC9VL3bg8f-ME:-OAcAasy4d7zbGNy-pzO:b3rc42i) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L28)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

I think you said you'd removed this option
</blockquote></details>

README updated to reflect this.
___
*[`README.md` line 35 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-8I1j87GQyJBe2om1:-OAcAdXt-O163_XgBPTO:bwngibl) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L35)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

in this image there's a user action "modify" but I thought this had been removed?
</blockquote></details>

Have updated the schema to remove this.
___
*[`README.md` line 108 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-6wZc2qVWImVELkDB:-OAcAfooE7REA-8kOBAR:bs44f8n) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L108)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

specify tag (for Readme could just be generic) ??
</blockquote></details>

README updated for this.
___
*[`README.md` line 113 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-74C85CwYolv1Eq6q:-OAcAjah1sx3I3DXiWsW:b7d1s2o) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L113)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

again specify tag?
</blockquote></details>

README updated.
___
*[`README.md` line 151 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8Ab9c8r-jJpC50tmY:-OAcAmBvCpT4r9gtk07j:b-8davfc) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L151)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

also version info for the panelapp API?
</blockquote></details>

README updated to include versions.
___
*[`README.md` line 152 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA-7QHe-quF09FgOPNV:-OAcAqWlD_ix5KslXRw_:bnufqgb) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/README.md#L152)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

is there any version information for these that should be noted here?
</blockquote></details>

Have also added versions used for these tools.
___
*[`app/bed_generator/api.py` line 207 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8E2B20cSKO7Y5LltS:-OAcAxVeDCCGEiba6K_b:b-4b4597) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/api.py#L207)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

not sure about this logic (not that the code is wrong, but that this isn't necessarily the most appropriate choice). I think if the transcript hasn't been provided and there's no MANE this needs to be a decision made by a clinical scientist. As a minimum I think any transcripts selected by this logic should be flagged?
</blockquote></details>

I have added a new field within the results that informs the user of logical decisions made in selecting transcripts. I have also added a check within the bed review process that informs the user that a clinical review must be made for any non-default (MANE) transcript selections within the BED file. User now has to review and tick a box accepting that this has been reviewed before submission will be accepted.
___
*[`app/bed_generator/api.py` line 370 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8IcZqCke_D3EmZ8jw:-OAcBaTBCg052o52zOuY:b-zexeiu) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/api.py#L370)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

what would this actually mean? how can a feature overlap "multiple unknown genes"?
</blockquote></details>

The only scenario that this will pick up is where a significantly large genomic region in coordinates is used as a query. This is very much an edge case, but the VEP overlap endpoint may return multiple "unknown" features if the region queried is large enough. I think the term "uncharacterised genomic regions" as opposed to genes is probably more appropriate and have updated to reflect this.
___
*[`app/bed_generator/api.py` line 413 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8JJY_-bMpIUa3Vp3k:-OAcDA5k1igh1ACWkwKv:bkhimhf) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/api.py#L413)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

does the user have the option to exclude amber and red genes?
</blockquote></details>

These are excluded from the frontend by default. The approach here is to retrieve the data in the backend to avoid performing secondary API calls if the user chooses to include amber/red genes.
___
*[`app/bed_generator/bed_generator.py` line 32 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA8KOdy-klIDG1ietdc:-OAcD_mtCwggXry04k0L:bds27fr) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/bed_generator.py#L32)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

Just want to check that this is fine for reverse strand genes, i.e. that elsewhere the code is handling that start and end in that context might be flipped so needs to be handled first?
</blockquote></details>

Have included screenshots within another reply - padding is applied based on strand.
___
*[`app/bed_generator/routes.py` line 95 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA9U-d052pX3eVCX7SB:-OAc40tuACQmaSmmTKil:b-bs2xq6) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/routes.py#L95)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

is this debug statement still needed?
</blockquote></details>

No longer needed and removed in latest commit.
___
*[`app/bed_generator/routes.py` line 120 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA9UGxn6Dk4INhMv4TY:-OAc5Aq469Ha1vCmoKzf:b4juhyv) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/routes.py#L120)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

same comment as before re padding- assume reverse strand gene start and stop coords are handled so Start isn't > than stop?
</blockquote></details>

Have adjusted the logic to simplify this and handle in routes. Forward and reverse are handled as you've described and have performed some double checks with a mixed range. Attaching image:

- Query using APOE (forward) and BRCA1 (reverse)
- 50bp padding has been applied to 5'.
- Padding applied discretely based on identifier strand direction within the same session

___
*[`app/bed_generator/routes.py` line 197 at r1](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OA9X2Rt04BnT890Ssec:-OAcAMEN2yrpCzEJTivh:bw0ridf) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/4237a033662c014660ef942a5bfd142e6ff2ca76/app/bed_generator/routes.py#L197)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

This is a very long try/except statement which is a bit hard to follow, especially with all the next ifs/fors. If it's working I'm happy for it to be left for now, but could an issue be added to the repo to consider reviewing/refining this in a future update?
</blockquote></details>

Agreed. I've refactored this down into smaller, (hopefully) easier to follow functions in relevant modules.

**Review by rebeccahaines1 — CHANGES_REQUESTED** [review]
> Reviewed 5 of 55 files at r1, 14 of 14 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)* status: all files reviewed, 5 unresolved discussions (waiting on @g-pyxl)

___
*[`app/bed_generator/panels.json` line 0 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OB-juVh9ZHQuON1IKxF:-OB-juVh9ZHQuON1IKxG:b-zb1neu) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_generator/panels.json#L0)):*
this is a dump of panelapp data. please can it be removed before this branch is merged in to main
___
*[`app/bed_generator/utils.py` line 234 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqxaZHAkaNgPPSwHM8:-OAqxaZHAkaNgPPSwHM9:bj09knl) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_generator/utils.py#L234)):*
> ```Python
>         panel_id: The ID of the panel.
>     """
>     genes = fetch_genes_for_panel(panel_id, include_amber=True, include_red=True)
> ```

is it clear to the user from the input page that amber and red genes will be includes and/or could/should there be an option to exclude them? Option to exclude can be a future development.
___
*[`app/bed_manager/routes.py` line 33 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqzHXu4bKXuYUzsnIM:-OAqzHXu4bKXuYUzsnIN:b-tlkoox) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_manager/routes.py#L33)):*
> ```Python
> @login_required
> def submit_bed_file():
>     # Logic for submitting a new BED file
> ```

? is something missing here?
___
*[`app/bed_manager/routes.py` line 39 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqzKnaAS76m_Xmx1C9:-OAqzKnaAS76m_Xmx1CA:b-yu9bza) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_manager/routes.py#L39)):*
> ```Python
> @login_required
> def review_bed_file(file_id):
>     # Logic for reviewing a BED file
> ```

see line 33
___
*[`app/static/js/results.js` line 12 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAwn2PhEZdIPXpEo_-S:-OAwn2PhEZdIPXpEo_-T:b-wfmz47) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/static/js/results.js#L12)):*
> ```JavaScript
>         settings = data;
>         // Set the default assembly in the dropdown
>         document.getElementById('genome-select').value = settings.defaultAssembly === 'GRCh38' ? 'hg38' : 'hg19';
> ```

? GRCh37 ?

**Review by g-pyxl — COMMENTED** [review]
Last commit will ideally remove as much redundancy as possible, and will update the README to include some end-to-end testing from this latest version (including using generated BEDs with some of our pipeline components to compare outputs).

> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-:-OB4nVqv6AFL-Qto8zIp:b-zd37t3)* status: all files reviewed, 5 unresolved discussions (waiting on @rebeccahaines1)

___
*[`app/bed_generator/panels.json` line  at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OB-juVh9ZHQuON1IKxF:-OB4kmlTD-BilC2DqvA9:bss4a2b) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_generator/panels.json#L)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

this is a dump of panelapp data. please can it be removed before this branch is merged in to main
</blockquote></details>

Removed.
___
*[`app/bed_generator/utils.py` line 234 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqxaZHAkaNgPPSwHM8:-OB4kp0P3VkVferI1GD6:b2ju29c) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_generator/utils.py#L234)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

is it clear to the user from the input page that amber and red genes will be includes and/or could/should there be an option to exclude them? Option to exclude can be a future development.
</blockquote></details>

They are excluded by default, but stored in the session query to prevent the need for additional API calls if they user chooses to include these. A default selection would look like this:



And when an option is ticked:



Unticking will remove these genes again.
___
*[`app/bed_manager/routes.py` line 33 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqzHXu4bKXuYUzsnIM:-OB4leT1CuV3lNkjdnXW:b-nbtfp) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_manager/routes.py#L33)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

? is something missing here?
</blockquote></details>

Old logic, this was changed and function no longer needed. Removed.
___
*[`app/bed_manager/routes.py` line 39 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqzKnaAS76m_Xmx1C9:-OB4lj1LCCVPp5jZLNfl:b-nbtfp) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_manager/routes.py#L39)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

see line 33
</blockquote></details>

Old logic, this was changed and function no longer needed. Removed.
___
*[`app/static/js/results.js` line 12 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAwn2PhEZdIPXpEo_-S:-OB4mWRg3_0Xu9FbctbJ:bsvkmak) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/static/js/results.js#L12)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

? GRCh37 ?
</blockquote></details>

This is the genome selection for IGV. The options available to IGV are 'hg38' and 'hg19'.

The code here is storing the initially query from the frontend as settings. If the assembly selected and passed from the initial query == "GRCh38", set IGV to 'hg38'. If anything else, set IGV to 'hg19' by default.

**Review by rebeccahaines1 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)* status: all files reviewed, 2 unresolved discussions (waiting on @g-pyxl)

___
*[`app/bed_manager/routes.py` line 33 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqzHXu4bKXuYUzsnIM:-OB5HaHu02sMfsQmjzdV:b8l9r1k) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_manager/routes.py#L33)):*
<details><summary><i>Previously, g-pyxl (George Doyle) wrote…</i></summary><blockquote>

Old logic, this was changed and function no longer needed. Removed.
</blockquote></details>

ok. please can you delete from the coed to avoid future confusion
___
*[`app/bed_manager/routes.py` line 39 at r3](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-OAqzKnaAS76m_Xmx1C9:-OB5HdM34yL8s4vtOCCo:b-a69b6x) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/02592a297500091669d61f73e7230f76f9e98283/app/bed_manager/routes.py#L39)):*
<details><summary><i>Previously, g-pyxl (George Doyle) wrote…</i></summary><blockquote>

Old logic, this was changed and function no longer needed. Removed.
</blockquote></details>

see line 33 above

**Review by rebeccahaines1 — APPROVED** [review]
> Reviewed 10 of 10 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @g-pyxl)

**Comment by g-pyxl** [comment]
Merged new changes from MVP branch.

**Review by rebeccahaines1 — CHANGES_REQUESTED** [review]
> Reviewed 5 of 7 files at r5, 3 of 8 files at r6, 39 of 39 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1)* status: all files reviewed, 5 unresolved discussions (waiting on @g-pyxl)

___
*[`app/bed_generator/transcript_processing.py` line 2 at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODWFdVU3QrNuB6fHD_9:-ODWFdVU3QrNuB6fHD_A:b6jk1he) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/transcript_processing.py#L2)):*
> ```Python
> """
> transcript_processing.py - Functions for processing transcript data.
> ```

I think all of these functions are duplicated from app/bed_generator/api.py
___
*[`app/bed_generator/panels.json` line 0 at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODW3qnrAToEePsjLh5C:-ODW3qnrAToEePsjLh5D:bae52tz) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/panels.json#L0)):*
I think this file needs deleting
___
*[`app/bed_generator/routes.py` line 151 at r6](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODM675oDOLkRNlM6FkM:-ODM675oDOLkRNlM6FkN:b-hfnh23) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/257db36ecfeb5b33a581dbfd6aeb3c59885e6eca/app/bed_generator/routes.py#L151)):*
> ```Python
>         # Check both original coordinates and rsID presence
>         is_variant = (
>             int(result['original_loc_start']) == int(result['original_loc_end']) or
> ```

is this assuming a variant is always a single base change? what about indels?
___
*[`app/bed_generator/api.py` line 4 at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODVr89eDpwoDAP7u5xx:-ODVr89fC4oa2RifOwmg:b-1gdd4t) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/api.py#L4)):*
> ```Python
> api.py - Fetches and processes variant information from Ensembl, TARK, and PanelApp APIs.
> 
> Functions:
> ```

this list is incomplete- please can it be reviewed
___
*[`app/bed_generator/genes.json` line 0 at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODW1vMf7dEb_3T42Zqj:-ODW1vMf7dEb_3T42Zqk:b-ll62h6) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/genes.json#L0)):*
Think this file needs to be deleted?

**Review by g-pyxl — COMMENTED** [review]
Main update accounts for padding to variants over 1bp. Various files removed, docstrings updated.

> *[Reviewable](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-:-ODk7mcX60-T7kD0_IS2:bytbfw0)* status: all files reviewed, 5 unresolved discussions (waiting on @rebeccahaines1)

___
*[`app/bed_generator/api.py` line 4 at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODVr89eDpwoDAP7u5xx:-ODk78Sx8jc1hRIi4IDF:byh4wlx) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/api.py#L4)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

this list is incomplete- please can it be reviewed
</blockquote></details>

Updated all module docstrings
___
*[`app/bed_generator/genes.json` line  at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODW1vMf7dEb_3T42Zqj:-ODk7GygB9MXWUwE6t_Y:bhofej8) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/genes.json#L)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

Think this file needs to be deleted?
</blockquote></details>

Deleted
___
*[`app/bed_generator/panels.json` line  at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODW3qnrAToEePsjLh5C:-ODk7HhICZIU32tZjjAN:bhofej8) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/panels.json#L)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

I think this file needs deleting
</blockquote></details>

Deleted
___
*[`app/bed_generator/routes.py` line 151 at r6](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODM675oDOLkRNlM6FkM:-ODk2Ylz08qipiJHtImn:bj3ypj9) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/257db36ecfeb5b33a581dbfd6aeb3c59885e6eca/app/bed_generator/routes.py#L151)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

is this assuming a variant is always a single base change? what about indels?
</blockquote></details>

I have applied the following handler for this in routes.py to account for longer variants. Tested using various indel identifiers, downloads and db entries both applying correct padding for these cases.     

# Handle SNPs

    if entry.get('is\_snp', False) or entry.get('rsid'):

        if snp\_padding:

            \# For multi-bp SNPs, apply padding to start and end directly

            if entry\['loc\_start'\] != entry\['loc\_end'\]:

                result\['loc\_start'\] = int(entry\['loc\_start'\]) - snp\_padding

                result\['loc\_end'\] = int(entry\['loc\_end'\]) + snp\_padding

            else:

                \# For single-bp SNPs, calculate from center point

                center = int(entry\['loc\_start'\])

                result\['loc\_start'\] = center - snp\_padding

                result\['loc\_end'\] = center + snp\_padding

        return result
___
*[`app/bed_generator/transcript_processing.py` line 2 at r7](https://reviewable.io/reviews/moka-guys/bedmakerGUI/1#-ODWFdVU3QrNuB6fHD_9:-ODk3LY192AvRSEFVnrr:b-obqsfc) ([raw file](https://github.com/moka-guys/bedmakerGUI/blob/cddc34d11bef614befc907667958c125f5686b77/app/bed_generator/transcript_processing.py#L2)):*
<details><summary><i>Previously, rebeccahaines1 wrote…</i></summary><blockquote>

I think all of these functions are duplicated from app/bed_generator/api.py
</blockquote></details>

Removed - was part of a refactoring branch and accidentally included
